kernel/kernel-api/src/main/java/io/delta/kernel/internal/snapshot/SnapshotManager.java

```diff
@@ -425,8 +431,7 @@ public Optional<LogSegment> getLogSegmentForVersion(
    * Helper function for the getLogSegmentForVersion above. Called with a provided files list, and
    * will then try to construct a new LogSegment using that.
    */
-  protected Optional<LogSegment> getLogSegmentForVersion(
-      Engine engine,
+  private LogSegment getLogSegmentForVersionHelper(
```

Comment:
I don't really like all these "getLogSegmentForVersionHelper" and "getLogSegmentForVersion" naming maybe we can update these to be more specific.

Like maybe this one can be `buildLogSegmentForVersion` or `constructLogSegment` since it isn't doing the listing?

Reply:
I totally agree. I'm just not sure if this is the PR to do that. I'm trying to do it bit by bit. Today, in master, we have

1. `getLogSegmentFrom(Engine engine, Optional<CheckpointMetaData> startingCheckpoint)`
2. `getLogSegmentForVersion(Engine engine, Optional<Long> startCheckpoint, Optional<Long> versionToLoad)`
3. `getLogSegmentForVersion(Engine engine,Optional<Long> startCheckpointOpt,Optional<Long> versionToLoadOpt, List<FileStatus> newFiles)`

Ideally, we would have only one method: `getLogSegmentForVersion(Engine engine, Optional<Long> versionToLoad)`

This PR gets rid of number 1. Number two is extensively used for tests.

Reply:
And i'm not sure if I love that number two is used for tests. We only have 2 entry points here: get snapshot for latest, get snapshot for version (from a public API perspective. note that get snapshot for timestamp really just used get snapshot for version).

So IMO our tests should be testing _that_ entry point, not testing some custom method where you can input a starting checkpoint, too. Ideally we just setup our tests better where we define what the starting checkpoint is via setting the table state, as opposed to having that as input to the method ....

Reply:
But that's a big refator 😅

Reply:
So I think in this PR i'd like to merge `getLogSegmentForVersion(Engine engine, Optional<Long> versionToLoad)` and in a followup I can go and remove the 33 test instances of `getLogSegmentForVersion(Engine engine, Optional<Long> startCheckpoint, Optional<Long> versionToLoad)`

Reply:
constructLogSegmentFromFileList